### PR TITLE
score importing

### PR DESCRIPTION
**Pull request by iturgeon**
* INCLUDES DB MIGRATIONS 
* new option added to get params for lti launch url to enable/disable score importing
* new GUI in module picker interface for enabling score importing (on by default)
* on student visit, the viewer state is populated with importable scores
* student view notifies user an import is possible on initial view and before starting an assessment
* once import is used, attempts are all used up and no further attempts are allowed
* importing is not allowed after the first attempt is completed
* splits viewer, editor, and assessment api into separate parts
* moves almost all db queries into model classes
* some updates to /dev page to include import options & organization

solves #228

### CODE REVIEW

File: packages/app/obojobo-express/server/lti.js [hunk 1]
```diff
@@ -717,6 +665,10 @@ const sendHighestAssessmentScore = (
 
 			result.outcomeServiceURL = outcomeData.serviceURL
 
+			// @TODO: move this above getRequiredDataForReplaceResult call
+			// @TODO: heck, why dont we just move all these throws down
+			// to the where the code that finds these issues occurs
+			// and pass them up using the normal throw/catch functionality
 			if (isPreview) {
 				throw Error(ERROR_PREVIEW_MODE)
```
Comment (qwertynerd97) [inline packages/app/obojobo-express/server/lti.js]: Was this supposed to have been moved in this PR, or is it an issue for a future PR?

File: packages/app/obojobo-module-selector/client/js/module-selector.js [hunk 2]
```diff
@@ -246,9 +255,21 @@ import '../css/module-selector.scss'
...
+			(allowScorePassback ? 'true' : 'false')
+		)
 	}
 
+	// utility
+	// function hasMoreItems() {
+	// 	const d = data
+	// 	return d.last < d.items.length
+	// }
+
```
Comment (qwertynerd97) [inline packages/app/obojobo-module-selector/client/js/module-selector.js]: I think this got accidentally left in

File: packages/obonode/obojobo-sections-assessment/server/attempt-end/attempt-import.js [hunk 3]
```diff
@@ -0,0 +1,117 @@
+const AssessmentScore = require('../models/assessment-score')
+const AssessmentModel = require('../models/assessment')
+const insertEvents = require('./insert-events')
+const lti = require('obojobo-express/server/lti')
+const db = require('obojobo-express/server/db')
+
+const attemptImport = async req => {
+	// @TODO validate req.body.importedAssessmentScoreId
```
Comment (qwertynerd97) [inline packages/obonode/obojobo-sections-assessment/server/attempt-end/attempt-import.js]: Do we want to validate in this PR or is this a separate issue?

**Comment by iturgeon** [comment]
Yep, good call.  Added a gatekeeper that tests if import has been used now (tests not written yet as I need sleep)

**Comment by iturgeon** [comment]
@zachberry found the source of the `undefined` error, fixed.

**Comment by ctcuff** [comment]
Noticed an issue regarding submissions, this was tested on the **first** attempt of an assessment:
1. **In a new tab**, open student course 2 w/ score importing
2. Take the assessment at the bottom but leave any number of questions blank (at least one needs to be blank)
3. Click submit but **leave the dialog open**
4. **In another new tab**, open the same student course. Normally, you should get a message that disallows having the same course open in two tabs, but if the submission dialog is still open, you get prompted to resume your attempt.
5. Close the **newer** tab you opened from step 4.
6. In the tab from step 1, click `Submit as incomplete`. You'll get a dialog that says: `currentVisit missing from request, got undefined`. Every time you try to take that same assessment in that tab, you'll get that same dialog, yet, you can still visit other pages. If you close and re-open the tab, you'll be prompted to resume your attempt

**Comment by iturgeon** [comment]
@zachberry squashed your bug. It was a variable naming issue in code that filters out all but the last unfinished attempt.

https://github.com/ucfopen/Obojobo/pull/1158/commits/f35b93619a5471fe187727c4bb829f99b212f5bf#diff-3a29544e8722527500fdfb688dbf60ac

**Comment by iturgeon** [comment]
Good eye on that problem @ctcuff I think it's a new issue.  I finally got to use the 'create an issue from this comment' button

**Comment by zachberry** [comment]
Closing in favor of #1461

**Pull request closed without merging**